src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -181,68 +170,85 @@ bool ThetaStarPathPlanner::updateVertex(CellCoordinate pCurr, CellCoordinate pNe
 }
 
 // top level function
-std::optional<std::vector<Point>> ThetaStarPathPlanner::findPath(const Point &start,
-                                                                 const Point &destination)
+std::variant<std::monostate, Curve, std::vector<Point>> ThetaStarPathPlanner::findPath(
+    const Point &start, const Point &dest, const Field &field,
+    const std::vector<Obstacle> &obstacles)
 {
-    CellCoordinate src, dest;
+    obstacles_ = obstacles;
+    std::variant<std::monostate, Curve, std::vector<Point>> retval;
+    CellCoordinate src_coord, dest_coord;
+
+
+    fieldXLength     = field.totalXLength();
+    fieldYLength     = field.totalYLength();
+    fieldXHalfLength = fieldXLength / 2.0;
+    fieldYHalfLength = fieldYLength / 2.0;
+
+    // initialize grid
+    numRows =
+        (int)((fieldXLength - ROBOT_MAX_RADIUS_METERS) / SIZE_OF_GRID_CELL_IN_METERS);
+    numCols =
+        (int)((fieldYLength - ROBOT_MAX_RADIUS_METERS) / SIZE_OF_GRID_CELL_IN_METERS);
```

Comment:
Correct me if I'm wrong, but won't this mean we always have a grid slightly smaller then the field?

Reply:
Yup. We take the total size of the field (includes the out of bounds area) and subtract a quarter of the robot diameter from each side to reduce bumping into the walls.

Reply:
Ah. I know somewhat out of scope, but could we leave a comment indicating as such?

Reply:
I'll tag it with #814 so I can go back and fix it